### PR TITLE
Fix displaying LHN instead Expensify uses Plaid to connect account 

### PR DESCRIPTION
**Pull request by shahinyan11**
### Details
Android - LHN is displayed instead Expensify uses Plaid to connect your account

### Fixed Issues
$ https://github.com/Expensify/App/issues/30617
PROPOSAL: https://github.com/Expensify/App/issues/30617#issuecomment-1789383290

### Tests
The steps only for Android: Native and IOS: Native
1. Go to account settings >> Workspace
2. Click on the workspace to view its settings
3. Click on Cards or Reimbursements settings
4. Select "Connect bank account" >> "Connect online with Plaid
5. Put app on background
6. Open app again
7. Verify that the last screen is visible and not LHN

- [x] Verify that no errors appear in the JS console

### Offline tests

The "Connect bank account" is not available for offline mode

### QA Steps
The steps only for Android: Native and IOS: Native
1. Go to account settings >> Workspace
2. Click on the workspace to view its settings
3. Click on Cards or Reimbursements settings
4. Select "Connect bank account" >> "Connect online with Plaid
5. Put app on background
6. Open app again
7. Verify that the last screen is visible and not LHN

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/46921547/82e8b28d-52a4-48b9-bbae-89af4c6443f4

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/46921547/eee3d704-5926-4965-87bb-da0a879e6c21

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/46921547/f11653fa-bbff-4935-8751-fcc771a18750

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/46921547/4b47de61-cb9b-4fd7-a7a7-3050fa38db25

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/46921547/684accd5-2214-4dd5-ac96-557633b6d279

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/46921547/dad00d8a-3092-4641-88fe-b81d54aefa2e

</details>

### CODE REVIEW

**Comment by s77rt** [comment]
Few notes:
- Please update the title to be more descriptive on what this PR is changing.
- The test steps are not complete, there should a step to verify against the expected behavior
- Can you please double check the offline test steps. I don't think Plaid won't work offline
- Add the missing of Screenshots/Videos for other platforms

**Comment by shahinyan11** [comment]
> * Can you please double check the offline test steps. I don't think Plaid won't work offline

@s77rt The connect bank account functionality is disabled for offline mode ( the  button is hidden ) . I just didn’t know whether to check it in this case or not .

And could you help me with this?  What did I miss doing 
Screenshot 2023-11-21 at 15 31 09

**Comment by s77rt** [comment]
Please fix the offline test steps. As for CLA check https://github.com/Expensify/App/pull/31614#issuecomment-1820717948

**Comment by shahinyan11** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by shahinyan11** [comment]
> Please fix the offline test steps. 

What do you mean by saying "fix the offline test steps". Should I remove the checkbox from the checklist or should I do something different ?

File: src/components/PlaidLink/index.native.js [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import {useEffect} from 'react';
-import {dismissLink, openLink, useDeepLinkRedirector, usePlaidEmitter} from 'react-native-plaid-link-sdk';
+import {dismissLink, openLink, usePlaidEmitter} from 'react-native-plaid-link-sdk';
 import Log from '@libs/Log';
 import CONST from '@src/CONST';
 import {plaidLinkDefaultProps, plaidLinkPropTypes} from './plaidLinkPropTypes';
 
 function PlaidLink(props) {
-    useDeepLinkRedirector();
```
Comment (s77rt) [inline src/components/PlaidLink/index.native.js:8]: `useDeepLinkRedirector` was removed from `react-native-plaid-link-sdk` since it's no longer needed.

Comment (shahinyan11) [reply src/components/PlaidLink/index.native.js:8]: Yes. There is not such hook in new version and it is not need. It is mentioned in plaid link official doc

**Comment by shahinyan11** [comment]
> Please add missing screenshots/videos

I added videos for both android and IOS . This case may not happen on other platforms because it related changing application state. Changing to background  and then returning to foreground is native app behavior only. 
If you do not agree with me, please write more precisely which screenshot or video is missing

**Comment by s77rt** [comment]
Each change should be tested on all the platforms. For the remaining platforms you can show that the Plaid flow is still working. You can do so by using the plaid sandbox info: username: user_good and password: pass_good

To use plaid sandbox make sure you are using the staging server i.e. go to Preference and enable the staging server switch

**Comment by shahinyan11** [comment]
> You can do so by using the plaid sandbox info: username: user_good and password: pass_good
> 
> To use plaid sandbox make sure you are using the staging server i.e. go to Preference and enable the staging server switch

I did not understand this well. How is `plaid sandbox` different from standard plaid with which I've been working with before. 
I apologize if I ask a lot of questions but please keep in mind that I’m new and don’t know a lot

**Comment by s77rt** [comment]
To add bank account using plaid you will have to supply real information. Or you can just use the plaid sandbox and use test credentials to add the bank account. 

(the process is the same, to use the sandbox you will just have to enable staging switch in preferences)

**Comment by shahinyan11** [comment]
> To add bank account using plaid you will have to supply real information. Or you can just use the plaid sandbox and use test credentials to add the bank account.

But I didn't authenticate in plaid account around this issue before. What steps I should do after auth in plaid account  ?

**Comment by shahinyan11** [comment]
Now should I change uploaded videos for android and IOS too. Or I can add such video only for other platforms ?

**Comment by s77rt** [comment]
Just add those videos for the missing platforms

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/16493223/69e5ed7f-9eae-4921-b7ea-f716189a26d3


</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/Expensify/App/assets/16493223/84cc28c5-206c-42f6-9bd0-215f6056fe3d


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/16493223/1f206bd6-a803-41ce-bdb4-30ebd435ee04


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/Expensify/App/assets/16493223/0fd9f7c2-7bd3-40ae-b276-2d8182f2dd67


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/Expensify/App/assets/16493223/3faa3f72-4fad-428f-81b8-26c2d8af7c66


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/16493223/cd29edb3-78d4-452e-8bd7-4da9b0c6d28a


</details>

**Comment by s77rt** [comment]
@shahinyan11 In the testings steps please add a note that the steps are only for native.

**Comment by shahinyan11** [comment]
I can't success to take video less than 10 mb for web versions. It always turns out a little more

**Comment by s77rt** [comment]
Encode it to a lower resolution

**Comment by shahinyan11** [comment]
I added video for MacOS: Chrome / Safari and then get this error 
Screenshot 2023-11-21 at 19 29 45

**Comment by shahinyan11** [comment]
@s77rt  Is there another account ?

**Comment by s77rt** [comment]
For plaid? No
But the lock is related to your Expensify account I think. Just try with another account

**Comment by shahinyan11** [comment]
@s77rt  I can't launch iOS: mWeb Safari and Android: mWeb Chrome . Can't run them using doing steps in documents

**Comment by shahinyan11** [comment]
@s77rt  There left only android mobWeb. But below error is always shown with or without `ngrok`.

**Comment by shahinyan11** [comment]
@s77rt  Шt finally worked out. Videos for all platforms are added

**Comment by s77rt** [comment]
Thanks!

**Review by s77rt — APPROVED** [review]

**Comment by shahinyan11** [comment]
> Thanks!

You to

**Review by robertjchen — DISMISSED** [review]
Looks great, thanks!

**Comment by robertjchen** [comment]
Looks like the Snyk service is complaining about `Out of sync package.json and package-lock.json detected. Please run "package.json install" and try again.` 🤔

**Comment by shahinyan11** [comment]
> Looks like the Snyk service is complaining about `Out of sync package.json and package-lock.json detected. Please run "package.json install" and try again.` 🤔

is it directed at me ? If yes I did not understand this  `run "package.json install"` . I tried ` npm install` but it didn't change anything

**Comment by s77rt** [comment]
@shahinyan11 You should not delete `package-lock.json`. Please keep the original one and only commit the changes after `npm install`

**Comment by shahinyan11** [comment]
> @shahinyan11 You should not delete `package-lock.json`. Please keep the original one and only commit the changes after `npm install`

But the `npm install` doesn't change anything so I can make a new commit

**Review by s77rt — APPROVED** [review]

**Comment by shahinyan11** [comment]
> But the `npm install` doesn't change anything and I can not make a new commit

@s77rt  You did not answer to this

**Comment by s77rt** [comment]
@shahinyan11 I see `package.json` changes are already included but if we are still out of sync then you should merge main into your branch then run npm install

**Comment by shahinyan11** [comment]
@s77rt  my repo is forked. And there is not any changes when I pull `main` branch. How I pull last updates from original `main` branch

**Comment by shahinyan11** [comment]
> @shahinyan11 I see `package.json` changes are already included but if we are still out of sync then you should merge main into your branch then run npm install

@s77rt I did it. But still the checking of license/synk is failed . Is it mean that something still wrong ?

**Comment by s77rt** [comment]
@shahinyan11 I don't know for sure but does not seem related. Robert will check and let you know if anything is needed from your side.

**Review by s77rt — APPROVED** [review]

**Review by robertjchen — APPROVED** [review]

**Comment by robertjchen** [comment]
Known issue with Snyk license check: https://expensify.slack.com/archives/C01GTK53T8Q/p1700498232891239

**Pull request merged**